platform/security/certificate/src/main/java/org/codice/ddf/certificate/PEMManager.java

```diff
@@ -0,0 +1,142 @@
+     * @return PrivateKey that was extracted from the PEM file
+     * -------------------------------------------------------------------
+     */
+    private static final Logger LOGGER = LoggerFactory.getLogger(PEMManager.class);
+
+    public static PrivateKey getPrivateKeyFromPEMFile(String fileName, String password) {
+        KeyPair kp;
+        PrivateKey privKey = null;
+        CertificateManager.registerBCSecurityProvider();
+        try {
+            File privateKeyFile = new File(fileName);
+            PEMParser pemParser = new PEMParser(new FileReader(privateKeyFile));
+            Object objectFromPEM = pemParser.readObject();
+            PEMDecryptorProvider decProv = new JcePEMDecryptorProviderBuilder()
+                    .build(password.toCharArray());
+            JcaPEMKeyConverter converter = new JcaPEMKeyConverter().setProvider("BC");
+            if (objectFromPEM instanceof PEMEncryptedKeyPair) {
+                kp = converter
+                        .getKeyPair(((PEMEncryptedKeyPair) objectFromPEM).decryptKeyPair(decProv));
+                privKey = kp.getPrivate();
+            } else if (objectFromPEM instanceof PrivateKeyInfo) {
+                PrivateKeyInfo pki = (PrivateKeyInfo) objectFromPEM;
+                privKey = converter.getPrivateKey(pki);
+            } else if (objectFromPEM instanceof PEMKeyPair) {
+                PrivateKeyInfo pki = ((PEMKeyPair) objectFromPEM).getPrivateKeyInfo();
+                privKey = converter.getPrivateKey(pki);
+            }
+        } catch (IOException e) {
+            LOGGER.error(e.getMessage(), e);
+        }
+
+        return privKey;
+    }
+
+    /**
+     * ---------------------------------------------------------------------
+     * Method: loadX509FromPEM
+     * <p/>
+     * Purpose:  Loads a X509 certificate from a valid PEM file that contains one
+     * <p/>
+     * Pre-condition:  BouncyCastle has been set as a security provider
+     * <p/>
+     * Post-condition: The X509Certificate contained in the PEM file is returned
+     *
+     * @param filePath path to the PEM file that contains the X509
+     * @return X509Certificate extracted from PEM file
+     * -------------------------------------------------------------------
+     */
+
+    public static X509Certificate loadX509FromPEM(String filePath) {
+        X509Certificate cert = null;
+        try {
+            JcaX509CertificateConverter certConverter = new JcaX509CertificateConverter()
+                    .setProvider("BC");
+            PEMParser pp = new PEMParser(new InputStreamReader(new FileInputStream(filePath)));
+            Object o = pp.readObject();
+            if (o instanceof X509CertificateHolder) {
+                X509CertificateHolder ch = (X509CertificateHolder) o;
+                cert = certConverter.getCertificate(ch);
+            } else {
+                LOGGER.error("File does not contain valid x509!");
+                throw new IOException();
```

Comment:
Don't log and throw. Just put the message inside the IOException and then it will be caught and logged. As it is, you're going to have 2 log messages that say more or less the same thing.